targets/XPlatCppSdk/source/test/TestApp/TestApp.cpp

```diff
@@ -16,106 +18,144 @@
+            &testRunner.suiteTestReport);
+
+        while (cloudResponse.empty())
+        {
+            std::this_thread::sleep_for(TimeValueMs(100));
+        }
+
+        // Publish the test summary (including cloud script response) to STDOUT.
+        Log("%s\n%s\n", testRunner.suiteTestSummary.c_str(), cloudResponse.c_str());
+
+        // Return 0 (success) if all tests passed. Otherwise, return 1 (error).
+        return testRunner.AllTestsPassed() ? 0 : 1;
     }
 
-    // Initialize the test runner/test data.
-    TestRunner testRunner;
-
-    // Add PlayFab API tests.
-    PlayFabApiTest pfApiTest;
-    pfApiTest.SetTitleInfo(testInputs);
-    testRunner.Add(pfApiTest);
-
-    PlayFabEventTest pfEventTest;
-    testRunner.Add(pfEventTest);
-
-    PlayFabTestMultiUserStatic pfMultiUserStaticTest;
-    testRunner.Add(pfMultiUserStaticTest);
-
-    PlayFabTestMultiUserInstance pfMultiUserInstanceTest;
-    testRunner.Add(pfMultiUserInstanceTest);
-
-    // Run the tests (blocks until all tests have finished).
-    testRunner.Run();
-
-    // Publish the test report via cloud script (and wait for it to finish).
-    PostResultsToCloud(testRunner.suiteTestReport);
-
-    while (cloudResponse.empty())
+    bool TestApp::LoadTitleData(TestTitleData& titleData)
     {
-        std::this_thread::sleep_for(TimeValueMs(100));
+        // Load JSON string in a platform-dependent way.
+        std::shared_ptr<char*> titleJsonPtr;
+        size_t size;
+        const bool loadedSuccessfully = LoadTitleDataJson(titleJsonPtr, size);
+
+        if (!loadedSuccessfully)
+            return false;
+
+        // Parse JSON string into output TestTitleData.
+        Json::CharReaderBuilder jsonReaderFactory;
+        Json::CharReader* jsonReader(jsonReaderFactory.newCharReader());
+        JSONCPP_STRING jsonParseErrors;
+        Json::Value titleDataJson;
+        const bool parsedSuccessfully = jsonReader->parse(*titleJsonPtr, *titleJsonPtr + size + 1, &titleDataJson, &jsonParseErrors);
+
+        if (parsedSuccessfully)
+        {
+            titleData.titleId = titleDataJson["titleId"].asString();
+            titleData.userEmail = titleDataJson["userEmail"].asString();
+            titleData.developerSecretKey = titleDataJson["developerSecretKey"].asString();
+        }
+
+        return parsedSuccessfully;
     }
 
-    // Publish the test summary (including cloud script response) to STDOUT.
-    printf("%s\n%s\n", testRunner.suiteTestSummary.c_str(), cloudResponse.c_str());
+    void TestApp::Log(const char* format, ...)
+    {
+        static char message[4096];
 
-    // Return 0 (success) if all tests passed. Otherwise, return 1 (error).
-    return testRunner.AllTestsPassed() ? 0 : 1;
-}
+        va_list args;
+        va_start(args, format);
+        _vsnprintf_s(message, sizeof(message), format, args);
+        va_end(args);
 
-static void OnPostReportError(const PlayFabError& error, void*)
```

Comment:
Wow, good call.... I thought we already deleted all the static test stuff... weird.